### ISSUE TITLE
Classify task?

### ISSUE THREAD

**Issue by tjasmin111** [WongKinYiu/YOLO#82]
Is there a plan to add a classify task, just like original YOLO?
I use both classify and detect task from the original yolov8 and would like to replace both with your repo.

**Comment by Abdul-Mukit** [comment]
I believe classification is one of the TODO tasks in this repo. Please check the landing readme page.

The original YOLO is YOLO1. Everything else including this repo are derivative works IMO. As far as I have read YOLOv5 is just a modified version of YOLO4. YOLOv8 is just a modified version of YOLOv9. I think this cycle will just keep continuing. 

Here are some helpful links that I learned from.
A review paper that reviews the different architectures v1 upto 2023. You can see that YOLOv5 and YOLOv8 didn't even have a paper.
https://arxiv.org/abs/2304.00501

Here is a good article from OpenCV that shows you different options and licenses. Old doc.
https://www.opencv.ai/blog/yolo-unraveled-a-clear-guide

Finally, a Redit post that talks about the APGL and how anything you output using APGL (weights, or anything else), has to be open souced unless you buy Enterprise license.
https://www.reddit.com/r/computervision/comments/1e3uxro/ultralytics_new_agpl30_license_exploiting/

**Comment by Abdul-Mukit** [comment]
You can consider starting the classify task? Maybe make a PR?
Shouldn't be very different. One just needs to reuse the backbone replace neck + head with a classification head. The do transfer learning I believe.

**Comment by tjasmin111** [comment]
I really like for it to have a classification task. But I think would be a bit hard for me to start the PR. Not very expert.

**Comment by Abdul-Mukit** [comment]
https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html
https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html
You can look at these tutorials to fameliarize yourself with the overall process.

**Comment by henrytsui000** [comment]
Hi,

I just added a classification head in commit d44dbc0. However, I haven't added the loss function, pretrained weight, or training schedule yet. For the dataset, I'll enable the dataloader to load Torchvision's CIFAR100 as a baseline. Also, check if the memory leak issue occurs with Torchvision's dataloader as well.

Best regards,  
Henry Tsui

**Comment by tjasmin111** [comment]
Good progress. I'll check it out. So it can be used to train from scratch not using a pre trained weight right?

**Comment by azrael05** [comment]
I will be raising a PR for the basic classification soon. You can use it your purpose. But I have limited resources so I won't be able to share the weights of imagenet training,
